packages/legend-application/src/stores/storage/WebApplicationClientSideStorageService.ts

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright (c) 2020-present, Goldman Sachs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  type PlainObject,
+  SerializationFactory,
+  usingModelSchema,
+} from '@finos/legend-shared';
+import { createModelSchema, list, primitive } from 'serializr';
+import type { GenericLegendApplicationStore } from '../ApplicationStore.js';
+import { WebApplicationClientSideStorage } from './WebApplicationClientSideStorage.js';
+
+export class StorageItem {
+  key!: string;
+  value!: string;
+
+  constructor(key: string, value: string) {
+    this.key = key;
+    this.value = value;
+  }
+
+  static create(key: string, value: string): StorageItem {
+    return new StorageItem(key, value);
+  }
+}
+
+const storageItemModelSchema = createModelSchema(StorageItem, {
+  key: primitive(),
+  value: primitive(),
+});
+
+export class StorageData {
+  items: StorageItem[] = [];
+
+  static readonly serialization = new SerializationFactory(
+    createModelSchema(StorageData, {
+      items: list(usingModelSchema(storageItemModelSchema)),
+    }),
+  );
+
+  static serialize(value: StorageData): PlainObject<StorageData> {
+    return StorageData.serialization.toJson(value);
+  }
+}
+
+export class WebApplicationClientSideStorageService {
```

Comment:
--> `StorageService`

Reply:
I think the way we should do this is to think of the storage as a place we load the initial value and the place where we explicitly persist the value every time it's changed. So we should have something like this:

e.g. wrapText

```ts
// initial load
this.wrapText = this.applicationStore.storageService.get(LEGEND_STUDIO_SETTINGS_KEY.EDITOR_WRAP_TEXT) ?? false;

// when user changes the value **explicitly**
// NOTE: I'm not so sure where should we do this though, we have a bunch of setter methods usually
// I think persistence should only occur when it's done by users directly
// so maybe calling persistence within theb ody of the setter in state is not ideal, but perhaps, the best place
// to invoke persistence is in the action/event handler method in the component 
this.applicationStore.storageService.persist(LEGEND_STUDIO_SETTINGS_KEY.EDITOR_WRAP_TEXT, this.wrapText);
```